programs/clearing_house/src/controller/bank_balance.rs

```diff
@@ -76,14 +130,51 @@ pub fn update_bank_balances(
     if let BankBalanceType::Borrow = update_direction {
         let deposit_token_amount =
             get_token_amount(bank.deposit_balance, bank, &BankBalanceType::Deposit)?;
+
         let borrow_token_amount =
             get_token_amount(bank.borrow_balance, bank, &BankBalanceType::Borrow)?;
 
         validate!(
             deposit_token_amount >= borrow_token_amount,
             ErrorCode::BankInsufficientDeposits,
             "Bank has insufficent deposits to complete withdraw"
-        )?
+        )?;
+    }
+
+    Ok(())
+}
+
+pub fn update_bank_balances_with_limits(
+    token_amount: u128,
+    update_direction: &BankBalanceType,
+    bank: &mut Bank,
+    bank_balance: &mut dyn BankBalance,
+) -> ClearingHouseResult {
+    update_bank_balances(token_amount, update_direction, bank, bank_balance)?;
+
+    let valid_withdraw = check_withdraw_limits(bank)?;
+
+    validate!(
+        valid_withdraw,
+        ErrorCode::BankDailyWithdrawLimit,
+        "Bank has hit daily withdraw limit"
+    )?;
+
+    Ok(())
+}
+
+pub fn check_bank_market_valid(
```

Comment:
tbh it's kinda weird that the bank check is based on the amm being updated, why'd you decide to do this instead of just doing the oracle validity check with the bank guardrail? easier to implement?

Reply:
yeah.. part of me wants to just put the oracle twap on the bank as well since there may be banks w/o perp markets and maybe linking spot market and bank makes more sense anyhow